cou/steps/plan.py

```diff
@@ -53,6 +54,147 @@
 logger = logging.getLogger(__name__)
 
 
+def pre_plan_sanity_checks(args: CLIargs, analysis_result: Analysis) -> None:
+    """Pre checks to generate the upgrade plan.
```

Comment:
I see little benefit on doing this. In `Java` for example, the checked exception forced the signature of every method would list all of the exceptions that it could pass to its caller and has a high price. If we start getting deep into this we could have a situation where we would need to go three or more levels checking all the exceptions that can happen.

On theory a change at a low level of the software would imply on updating on the higher level and I don't think it worth.

Reply:
@samuelengineer that's good point, I lost myself with warlock operators in this PR. 
@gabrielcocenza I see your point, however all those `verify_*` function are simple helper functions, and we expected this function to raise those exceptions. If I look at the function name `pre_plan_sanity_checks`, and then the description, I could not tell what this function actually do? It's not returning anything and not mention raising anything.

Please stop referring to that book (I'm going to buy and read it, because there could be good point). I believe that you're misinterpreting things from it, or it has written not for Python. Thanks